### PR TITLE
[nicklist][#486] add patch 0.4.6+

### PR DESCRIPTION
**Pull request by bw1**
@coekie 
https://github.com/coekie/irssi-scripts/blob/master/nicklist.pl
http://wouter.coekaerts.be/tmp/nicklist.pl

  * Add support for 'other' modes (&~,...)
  * Give different (configurable) colors to friends, dependant on flags.
  * Fix another screen sizing issue

### CODE REVIEW

File: scripts/nicklist.pl [hunk 1]
```diff
@@ -58,25 +75,47 @@ sub cmd_help {
...
 
-	($prefix_mode[$MODE_OP] = Irssi::settings_get_str('nicklist_prefix_mode_op')) =~ s/\\e/\033/g;
-	($prefix_mode[$MODE_HALFOP] = Irssi::settings_get_str('nicklist_prefix_mode_halfop')) =~ s/\\e/\033/g;
-	($prefix_mode[$MODE_VOICE] = Irssi::settings_get_str('nicklist_prefix_mode_voice')) =~ s/\\e/\033/g;
-	($prefix_mode[$MODE_NORMAL] = Irssi::settings_get_str('nicklist_prefix_mode_normal')) =~ s/\\e/\033/g;
-	
+	($prefix_mode{'@'} = Irssi::settings_get_str('nicklist_prefix_mode_op')) =~ s/\\e/\033/g;
+	($prefix_mode{'%'} = Irssi::settings_get_str('nicklist_prefix_mode_halfop')) =~ s/\\e/\033/g;
+	($prefix_mode{'+'} = Irssi::settings_get_str('nicklist_prefix_mode_voice')) =~ s/\\e/\033/g;
+	($prefix_mode{' '} = Irssi::settings_get_str('nicklist_prefix_mode_normal')) =~ s/\\e/\033/g;
```
Comment (dequis) [inline scripts/nicklist.pl]: This seems like a downgrade to me

Comment (ailin-nemui) [reply scripts/nicklist.pl]: in which way?

Comment (dequis) [reply scripts/nicklist.pl]: We're merging some externally maintained version that diverged from this one, right? This particular change looks like it increased the hardcodedness, going from $MODE_OP to '@'.

File: scripts/nicklist.pl [hunk 2]
```diff
@@ -222,19 +264,14 @@ sub screen_size {
...
 		($row, $col, $xpixel, $ypixel) = unpack('S4', $winsize);
 	}
-	
+
 	# set screen width
 	$irssi_width = $col-$nicklist_width-1;
 	$height = $row-1;
-	
-	# on some recent systems, "screen -X fit; screen -X width -w 50" doesn't work, needs a sleep in between the 2 commands
-	# so we wait a second before setting the width
```
Comment (dequis) [inline scripts/nicklist.pl]: Changing this to the simpler method also seems like a downgrade

File: scripts/nicklist.pl (hunk 1 above)
Comment (ailin-nemui) [reply scripts/nicklist.pl]: ah yea but the externally diverged version is from the original author

File: scripts/nicklist.pl (hunk 2 above)
Comment (ailin-nemui) [reply scripts/nicklist.pl]: true, it looks odd. but seems to be what the author wanted --> https://github.com/coekie/irssi-scripts/commit/6d127c24b5648c7b51c3e85926c7804e0234ce4d#diff-7753d7712b1505f36887cc9678217810L201

**Comment by bw1** [comment]
nl

**Comment by dequis** [comment]
@bw1 :heart:

**Pull request merged**